From 0d19f19be9a93642f10b4c039aafd9e62f35e8fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Thu, 17 Mar 2022 10:16:30 +0100
Subject: [PATCH] shared/install: stop passing duplicate root argument to
 install_name_printf()

All callers were just passing info + info->root, we can simplify this.

(cherry picked from commit 38e8a6c7fdffd3389cb3596139b1309579193946)

Related: #2082131
---
 src/shared/install-printf.c   | 7 +++----
 src/shared/install-printf.h   | 3 +--
 src/shared/install.c          | 8 ++++----
 src/test/test-load-fragment.c | 2 +-
 4 files changed, 9 insertions(+), 11 deletions(-)

diff --git a/src/shared/install-printf.c b/src/shared/install-printf.c
index 963102674b..7aad1b7443 100644
--- a/src/shared/install-printf.c
+++ b/src/shared/install-printf.c
@@ -105,9 +105,8 @@ static int specifier_last_component(char specifier, const void *data, const char
 
 int install_name_printf(
                 UnitFileScope scope,
-                const UnitFileInstallInfo *i,
+                const UnitFileInstallInfo *info,
                 const char *format,
-                const char *root,
                 char **ret) {
         /* This is similar to unit_name_printf() */
 
@@ -124,9 +123,9 @@ int install_name_printf(
                 {}
         };
 
-        assert(i);
+        assert(info);
         assert(format);
         assert(ret);
 
-        return specifier_printf(format, UNIT_NAME_MAX, table, root, i, ret);
+        return specifier_printf(format, UNIT_NAME_MAX, table, info->root, info, ret);
 }
diff --git a/src/shared/install-printf.h b/src/shared/install-printf.h
index d2cccdf66d..60d3a9fc55 100644
--- a/src/shared/install-printf.h
+++ b/src/shared/install-printf.h
@@ -6,7 +6,6 @@
 
 int install_name_printf(
                 UnitFileScope scope,
-                const UnitFileInstallInfo *i,
+                const UnitFileInstallInfo *info,
                 const char *format,
-                const char *root,
                 char **ret);
diff --git a/src/shared/install.c b/src/shared/install.c
index a864039f44..f911d527df 100644
--- a/src/shared/install.c
+++ b/src/shared/install.c
@@ -1205,7 +1205,7 @@ static int config_parse_also(
                 if (r == 0)
                         break;
 
-                r = install_name_printf(ctx->scope, info, word, info->root, &printed);
+                r = install_name_printf(ctx->scope, info, word, &printed);
                 if (r < 0)
                         return log_syntax(unit, LOG_WARNING, filename, line, r,
                                           "Failed to resolve unit name in Also=\"%s\": %m", word);
@@ -1254,7 +1254,7 @@ static int config_parse_default_instance(
                 return log_syntax(unit, LOG_WARNING, filename, line, 0,
                                   "DefaultInstance= only makes sense for template units, ignoring.");
 
-        r = install_name_printf(ctx->scope, info, rvalue, info->root, &printed);
+        r = install_name_printf(ctx->scope, info, rvalue, &printed);
         if (r < 0)
                 return log_syntax(unit, LOG_WARNING, filename, line, r,
                                   "Failed to resolve instance name in DefaultInstance=\"%s\": %m", rvalue);
@@ -1850,7 +1850,7 @@ static int install_info_symlink_alias(
         STRV_FOREACH(s, info->aliases) {
                 _cleanup_free_ char *alias_path = NULL, *dst = NULL, *dst_updated = NULL;
 
-                q = install_name_printf(scope, info, *s, info->root, &dst);
+                q = install_name_printf(scope, info, *s, &dst);
                 if (q < 0) {
                         unit_file_changes_add(changes, n_changes, q, *s, NULL);
                         r = r < 0 ? r : q;
@@ -1940,7 +1940,7 @@ static int install_info_symlink_wants(
         STRV_FOREACH(s, list) {
                 _cleanup_free_ char *dst = NULL;
 
-                q = install_name_printf(scope, info, *s, info->root, &dst);
+                q = install_name_printf(scope, info, *s, &dst);
                 if (q < 0) {
                         unit_file_changes_add(changes, n_changes, q, *s, NULL);
                         return q;
diff --git a/src/test/test-load-fragment.c b/src/test/test-load-fragment.c
index 9df53cec2b..a87c654f4e 100644
--- a/src/test/test-load-fragment.c
+++ b/src/test/test-load-fragment.c
@@ -531,7 +531,7 @@ TEST(install_printf, .sd_booted = true) {
                 _cleanup_free_ char *t = NULL,                          \
                         *d1 = ASSERT_PTR(strdup(i.name)),               \
                         *d2 = ASSERT_PTR(strdup(i.path));               \
-                int r = install_name_printf(scope, &src, pattern, NULL, &t); \
+                int r = install_name_printf(scope, &src, pattern, &t);  \
                 assert_se(result ? r >= 0 : r < 0);                     \
                 memzero(i.name, strlen(i.name));                        \
                 memzero(i.path, strlen(i.path));                        \
